app/channel.go

```diff
@@ -223,6 +241,24 @@ func (a *App) CreateChannel(channel *model.Channel, addMember bool) (*model.Chan
 		})
 	}
 
+	esInterface := a.Elasticsearch
+	if esInterface != nil && *a.Config().ElasticsearchSettings.EnableIndexing {
+		if sc.Type == "O" {
```

Comment:
Why this conditional check?

Reply:
My understanding is that autocompletion only happens for public or open channels, so this check exists to only index the open ones.

Reply:
Oh yeah, seems you are right (although I'm quite surprised by that!).